elasticdl/python/worker/worker.py

```diff
@@ -513,9 +518,17 @@ def _process_save_model_task_if_needed(self):
         task, dataset = (
             self._task_data_service.get_save_model_task_and_dataset()
         )
-        if task and dataset:
+        if (task is not None) and (dataset is not None):
```

Comment:
`if task and not dataset` is more concise.

Reply:
For SaveModel task, the dataset is necessary to build the model, so we need both task and dataset.